rllib/env/vector_env.py

```diff
@@ -325,6 +377,18 @@ def try_reset(self, env_id: Optional[EnvID] = None) -> MultiEnvDict:
             else 0: {_DUMMY_AGENT_ID: self.vector_env.reset_at(env_id)}
         }
 
+    @override(BaseEnv)
+    def try_restart(self, env_id: Optional[EnvID] = None) -> None:
+        assert env_id is None or isinstance(env_id, int)
+        # Restart the sub-env at the index.
+        self.vector_env.restart_at(env_id)
+        # Auto-reset (get ready for next `poll()`).
+        self.new_obs[env_id] = self.vector_env.reset_at(env_id)
+        # Reset all other states to null values.
+        self.cur_rewards[env_id] = None
+        self.cur_dones[env_id] = False
+        self.cur_infos[env_id] = None
```

Comment:
same thing. can we create a func to share these per-env initialization, so they don't get out-of-date easily.

Reply:
done